ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -57,11 +57,22 @@ message Commands {
   // Required
   repeated Command commands = 8;
 
-  // The length of the time window during which all commands with the same ``act_as`` parties and
-  // the same command ID will be deduplicated.
-  // Duplicate commands submitted before the end of this window return an ALREADY_EXISTS error.
-  // Optional
-  google.protobuf.Duration deduplication_time = 9;
+  oneof deduplication {
+
+      // Specifies the length of the deduplication period.
```

Comment:
Good point regarding this change. If you check the future lines it says:

```
    // The participant or Daml ledger converts the duration into a timestamp ``deduplication_start``
      // by subtracting the duration from its local clock at some point during processing the submission.
```


So the deduplication is calculated in the past for the new version, but in the future for the old version. How do we want to capture this in the documentation? @SamirTalwar-DA @fabiotudone-da

Reply:
In the proposal PR https://github.com/digital-asset/daml/pull/9567/files#diff-ed4ed30e4f593aaf9d013b39ed5272dfc0df08c548cc78f110fd4471eaa6f59cR72-R95, we decided to not reuse the existing field for the deduplication_duration and instead introduce a new field, so that we can then later properly deprecate the field and issue warnings if someone still uses it.

I'm against documenting the future aspect of this field. For backwards-compatibility, we want to emulate the old behaviour by treating the future-looking period as if it was backward-looking. If an application specifies the same deduplication time for each repetition of its commands, then it won't notice a difference. If it uses different times for different resubmissions, it may notice a difference, but @bame-da agreed that command dedup is currently sufficiently broken to warrant such a breaking change. Otherwise, if we document "future-looking" here, then all applications will break as soon as they switch to a Read/WriteService-v2 implementation because that would then have to reject deduplication_time specifications.